src/main/java/com/redhat/devtools/lsp4ij/features/documentation/TextMateHighlighterHelper.java

```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.lsp4ij.features.documentation;
+
+import com.intellij.openapi.editor.colors.EditorColorsManager;
+import com.intellij.openapi.editor.colors.EditorColorsScheme;
+import com.intellij.openapi.editor.ex.util.DataStorage;
+import com.intellij.openapi.editor.ex.util.LexerEditorHighlighter;
+import com.intellij.openapi.editor.richcopy.HtmlSyntaxInfoUtil;
+import com.intellij.openapi.fileTypes.SyntaxHighlighter;
+import com.intellij.openapi.util.registry.Registry;
+import com.redhat.devtools.lsp4ij.LanguageServersRegistry;
+import com.redhat.devtools.lsp4ij.internal.StringUtils;
+import com.vladsch.flexmark.html.HtmlWriter;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.plugins.textmate.TextMateService;
+import org.jetbrains.plugins.textmate.language.TextMateLanguageDescriptor;
+import org.jetbrains.plugins.textmate.language.syntax.highlighting.TextMateHighlighter;
+import org.jetbrains.plugins.textmate.language.syntax.lexer.TextMateHighlightingLexer;
+import org.jetbrains.plugins.textmate.language.syntax.lexer.TextMateLexerDataStorage;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Helper to highlight code with the proper textmate.
+ */
+public class TextMateHighlighterHelper {
+
+    private static final Map<String, String> languageFileExtensionCache;
+
+    static {
+        languageFileExtensionCache = new HashMap<>();
+        // TODO: fill this mapping in an extension point
+        // languageFileExtensionCache.put("python", "py");
+        // languageFileExtensionCache.put("typescript", "ts");
+        // languageFileExtensionCache.put("clojure", "clj");
+        // languageFileExtensionCache.put("rust", "rs");
+    }
+
+    public static boolean highlightWithTextMate(CharSequence code, final String language, HtmlWriter html) {
+        // Find TextMate language descriptor by file extension.
+        var languageDescriptor = findLanguageDescriptor(language);
+        if (languageDescriptor != null) {

```

Comment:
I would return early, just for code readablity

```
if (languageDescriptor == null) {
    return false;
}
```

Reply:
fixed